File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>
//Declerations
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string input = get_string("Text: ");
    int letters = count_letters(input);
    int words = count_words(input);
    int sentences = count_sentences(input);

    float L = (letters / (words / 100.0));
    //printf("%f\n", L);
    float S = (sentences / (words / 100.0));
    //printf("%f\n", S);
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }



    // printf("Letters: %i\n",letters);
    // printf("Words: %i\n",words);
    // printf("Sentences: %i\n",sentences);
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    // Count starts with 1 since there is no space before first word.
    int count = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == 32)
        {
            count++;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            count++;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Getting input
    text = get_string("Text: ")
    # Calculating letters,words and sentences
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    # Calculating Coleman-Liau index
    L = (letters / (words / 100.0))
    S = (sentences / (words / 100.0))
    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)
    # Posting grade with control
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def count_letters(text):
    count = 0
    for i in text:
        if i.isalpha():
            count += 1
    return count


def count_words(text):
    count = 1
    for i in text:
        if i.endswith(" "):
            count += 1
    return count


def count_sentences(text):
    count = 0
    for i in text:
        if i.endswith(".") or i.endswith("?") or i.endswith("!"):
            count += 1
    return count


if __name__ == "__main__":
    main()